example/tests/example_test.py

```
@@ -1,22 +1,60 @@
 import unittest
 
-from snapstack import Runner
+from snapstack import Plan, Step, Setup, Cleanup
 
 
 class ExampleTest(unittest.TestCase):
 
-    def test(self):
-        r = Runner(
+    def test_example(self):
+        '''
+        _test_example_
+
+        Basic test, which installs and tests and example
+        snap, after letting the Plan object build the default
+        snapstack.
+
+        Note: this will install snapstack on your system!
+
+        '''
+        example = Step(
             snap='example',  # Name of the snap
-            location='./tests/',  # Parent location for tests and files
-                                  # below. {local} is a magic string that
-                                  # points at the 'tests' directory.
+            script_loc='./tests/',  # Parent location for tests and files
+                                    # below. {local} is a magic string that
+                                    # points at the 'tests' directory.
             tests=['example.sh'],  # Test the snap
             files=['example.json'],  # Config files
-            base=None  # Don't override default Base.
+            snap_store=False  # Build from local source, rather than
+                              # install from the snap store.
         )
+        plan = Plan(tests=[example])
+        plan()
+
+    def test_custom_base(self):
+        '''
+        _test_custom_base_
+
+        Reference code for adding and removing stuff from our Setup
+        and Cleanup.
+
+        '''
+        setup = Setup()
+        cleanup = Cleanup()
+
+        # Add a step.
+        example = Step(
+            snap='example',
+            script_loc='./tests/',
+            tests=['example.sh'],
+            files=['example.json'],
+        )
+        setup.add_steps(('example', example))
+
+        # Remove steps.
+        setup.remove_steps(
+            'keystone', 'nova', 'neutron', 'glance', 'nova_hypervisor',
+            'neutron_ext_net')
```

I think without cleanup modifications, this will try to cleanup keystone nova neutron etc from the DB. But maybe those fail silently and it's ok.